### PR TITLE
pin: use separate dagservice for storing pinsets

### PR DESCRIPTION
**Pull request by whyrusleeping**
License: MIT
Signed-off-by: Jeromy <why@ipfs.io>

### CODE REVIEW

File: core/builder.go [hunk 1]
```diff
@@ -171,13 +171,15 @@ func setupNode(ctx context.Context, n *IpfsNode, cfg *BuildCfg) error {
 
 	n.Blocks = bserv.New(n.Blockstore, n.Exchange)
 	n.DAG = dag.NewDAGService(n.Blocks)
-	n.Pinning, err = pin.LoadPinner(n.Repo.Datastore(), n.DAG)
+
+	internalDag := dag.NewDAGService(bserv.New(n.Blockstore, offline.Exchange(n.Blockstore)))
```
Comment (Kubuxu) [inline core/builder.go]: We are using offline DAG service in many plances, maybe we should create helper function for that?

Comment (whyrusleeping) [inline core/builder.go]: yeah, thats probably a good idea

**Comment by Kubuxu** [comment]
LGTM

**Comment by whyrusleeping** [comment]
Yeah, lets do that refactor later. Its going to touch a lot of files and i prefer to keep PRs focused

**Pull request merged**